.circleci/config.yml

```diff
@@ -1,56 +1,151 @@
+version: 2
+
 defaults: &defaults
   working_directory: /routemaster
   docker:
-    - image: deliveroo/circleci:0.1.0
+    - image: deliveroo/circleci:0.1.8
 
-version: 2
+build_steps: &build_steps
+  steps:
+    - setup_remote_docker:
+        reusable: true
+        version: 17.05.0-ce
 
-jobs:
-  run_test_suite:
-    <<: *defaults
+    - checkout
 
-    # we want to run tests multiple times to minimise the risk of flaky tests in
-    # this (critical) application
-    parallelism: 5
+    - run:
+        name: Clean up env
+        command: ci down
+
+    - run:
+        name: Build CI Image
+        command: |
+          docker build \
+            -f Dockerfile \
+            --build-arg PROCESS=${ROUTEMASTER_PROCESS}
+            -t "${CIRCLE_PROJECT_REPONAME}-${ROUTEMASTER_PROCESS}:${CIRCLE_SHA1}" \
+            .
+
+push_steps: &push_steps
+  steps:
+  - setup_remote_docker:
+      reusable: true
+      version: 17.05.0-ce
+
+  - run:
+      name: Push to Image Repository
+      command: |
+        `print_env ${TARGET}`
+        push_image_to_ecr \
```

Comment:
☣️ ☣️ ☣️ Here be dragons. ☣️ ☣️ ☣️

Please export the artifact between steps because they may be ran on different machines.

Reply:
Right. I'll do that. Thanks.